sdk/keyvault/keyvault-secrets/README.md

```diff
@@ -215,18 +217,41 @@ _deleted_ secret. A deleted secret can't be updated. They can only be either
 read, recovered or purged.
 
 ```javascript
-await client.deleteSecret(secretName);
+const poller = await client.beginDeleteSecret(secretName)
+await poller.pollUntilDone();
 
 // If soft-delete is enabled, we can eventually do:
 await client.getDeletedSecret(secretName);
-// Deleted secrets can also be recovered or purged:
-await client.recoverDeletedSecret(secretName);
-// await client.purgeDeletedSecret(secretName);
+
+// Deleted secrets can also be recovered or purged.
+
+// recoverDeletedSecret returns a poller, just like beginDeleteSecret.
+const recoverPoller = await client.beginRecoverDeletedSecret(secretName)
+const recoverPoller.pollUntilDone();
+
+// And then, to purge the deleted secret:
+await client.purgeDeletedSecret(secretName);
 ```
 
-Since the deletion of a secret won't happen instantly, some time is needed
-after the `deleteSecret` method is called before the deleted secret is
-available to be read, recovered or purged.
+Since Secrets take some time to get fully deleted, `beginDeleteSecret`
+returns a Poller object that keeps track of the underlying Long Running
+Operation according to our guidelines:
+https://azure.github.io/azure-sdk/typescript_design.html#ts-lro
+
+
+Once you receive the poller, you'll be able to either run individual service
+calls until the secret is deleted, or wait until the process is done:
```

Comment:
This should call out that it's not necessary unless you plan to force purge immediately.